### PR TITLE
Remove generated NewXWithDefaults methods and related documentation

### LINKED ISSUE #194: `New{{classname}}WithDefaults` functions... do we want them?
Currently, for every model struct, we generate a `New{{classname}}WithDefaults` function.

For `Response` models... this seems trivially unnecessary, as response models only exist to deserialize data from the server.

For `Request` models... I wonder if this is **also** unnecessary, because:

* All of the default values that we know about, are declared in the OpenAPI

* They are declared in the OpenAPI, because they are defined as defaults in `sdk/framework.Path` structs

* The Vault server request handling code is going to automatically apply these defaults, when the given parameters are absent from requests sent over the network

* So what is the point in providing extra methods which help clients of the API fill in the very same defaults the server side is going to fill in anyway?

Unless I'm missing something here, I wonder if we might just delete this chunk of the template entirely:

https://github.com/hashicorp/vault-client-go/blob/21d51ea1c54800bd4579fea8ab0f5bf26cf62a4e/generate/templates/model_simple.handlebars#L23-L33

### PR DESCRIPTION
**Pull request by averche**
## Description

Removing New__WithDefaults generated methods since they are not particularly useful and somewhat confusing as explained in #194.

Also cleaning up some of the old generated documentation parts within the model docs.

Resolves #194

## How has this been tested?

Relying on CI.

### CODE REVIEW

**Comment by dhuckins** [comment]
are there any files that are worth taking a look at? (1,115 is a lot!)

**Comment by averche** [comment]
The changes are only in the three `generate/templates/*.handlebars` files. The rest are generated but should give you an idea on what's changed :)

File: generate/templates/model_doc.handlebars [hunk 1]
```diff
@@ -10,77 +10,6 @@ Name | Type | Description | Notes
 {{#each vars}}**{{name}}** | {{#unless required}}Pointer to {{/unless}}{{#if isContainer}}{{#if isArray}}{{#with items}}{{#unless isPrimitiveType}}{{#unless isFile}}[{{/unless}}{{/unless}}**[]{{dataType}}**{{#unless isPrimitiveType}}{{#unless isFile}}]({{dataType}}.md){{/unless}}{{/unless}}{{/with}}{{/if}}{{#if isMap}}{{#with items}}{{#unless isPrimitiveType}}{{#unless isFile}}[{{/unless}}{{/unless}}**map[string]{{dataType}}**{{#unless isPrimitiveType}}{{#unless isFile}}]({{#unless baseType}}{{dataType}}{{/unless}}{{baseType}}.md){{/unless}}{{/unless}}{{/with}}{{/if}}{{/if}}{{#unless isContainer}}{{#unless isPrimitiveType}}{{#unless isFile}}{{#unless isDateTime}}[{{/unless}}{{/unless}}{{/unless}}**{{dataType}}**{{#unless isPrimitiveType}}{{#unless isFile}}{{#unless isDateTime}}]({{#unless baseType}}{{dataType}}{{/unless}}{{baseType}}.md){{/unless}}{{/unless}}{{/unless}}{{/unless}} | {{#if description}}{{description}}{{/if}} | {{#unless required}}[optional] {{/unless}}{{#if isReadOnly}}[readonly] {{/if}}{{#with defaultValue}}[default to {{{.}}}]{{/with}}
 {{/each}}
 {{/unless}}
-
-## Methods
```
Comment (dhuckins) [inline generate/templates/model_doc.handlebars:14]: wouldn't we still need a `Methods` section?

Comment (averche) [reply generate/templates/model_doc.handlebars:14]: These had previously been removed in the actual models ([example](https://github.com/hashicorp/vault-client-go/blob/main/schema/model_alias_create_request.go)). Here I'm just cleaning them up the docs to match the code.

**Review by dhuckins — APPROVED** [review]

**Pull request merged**